sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetIOSuite.scala

```diff
@@ -1060,6 +1061,25 @@ class ParquetIOSuite extends QueryTest with ParquetTest with SharedSparkSession
       }
     }
   }
+
+  test("SPARK-36825: year-month/day-time intervals written and read as INT32/INT64") {
+    Seq(
+      YearMonthIntervalType() -> ((i: Int) => Period.of(i, i, 0)),
+      DayTimeIntervalType() -> ((i: Int) => Duration.ofDays(i).plusSeconds(i))
+    ).foreach { case (it, f) =>
+      val data = (1 to 10).map(i => Row(i, f(i)))
+      val schema = StructType(List(StructField("d", IntegerType, false),
+        StructField("i", it, false)).toArray)
```

Comment:
```suggestion
      val schema = StructType(Array(StructField("d", IntegerType, false),
        StructField("i", it, false)))
```

Reply:
haha, I actually borrowed the code from your test https://github.com/apache/spark/pull/19702/files#diff-07f918aa5538eb47de4ef8065cf016e773bb472466dbef73898c1517272db191R244-R245 :-)